File: src/Java/ImplicitFunctions/ImplicitDataSet.java
import vtk.vtkRTAnalyticSource;
import vtk.vtkImplicitDataSet;
import vtk.vtkNativeLibrary;

class ImplicitDataSet{  
	
	static {
	    if (!vtkNativeLibrary.LoadAllNativeLibraries()) {
	      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) {
	        if (!lib.IsLoaded()) {
	          System.out.println(lib.GetLibraryName() + " not loaded");
	        }
	      }
	    }
	    vtkNativeLibrary.DisableOutputWindow(null);
	  }
	
	
    public static void main(String args[]){  
    	
    		  vtkRTAnalyticSource waveletSource = new vtkRTAnalyticSource();
    		  waveletSource.Update();
    		  
    		  vtkImplicitDataSet implicitWavelet = new vtkImplicitDataSet();
    		  implicitWavelet.SetDataSet(waveletSource.GetOutput());

    		  double x[] = new double[] {0.5,0,0};
    		  // Value should roughly be 258.658.

    		 
		  System.out.print("x: " + implicitWavelet.EvaluateFunction(x));
		  
    }   
}   

File: src/Cxx/ImplicitFunctions/ImplicitDataSet.cxx
#include <vtkImageData.h>
#include <vtkImplicitDataSet.h>
#include <vtkRTAnalyticSource.h>
#include <vtkSmartPointer.h>
#include <vtkSphereSource.h>

int main(int /* argc */, char * /* argv */[]) {
  vtkSmartPointer<vtkRTAnalyticSource> waveletSource =
      vtkSmartPointer<vtkRTAnalyticSource>::New();
  waveletSource->Update();

  vtkSmartPointer<vtkImplicitDataSet> implicitWavelet =
      vtkSmartPointer<vtkImplicitDataSet>::New();
  implicitWavelet->SetDataSet(waveletSource->GetOutput());

  double x[3] = {.5, 0, 0};
  // Value should roughly be 258.658.
  cout << "x: " << implicitWavelet->EvaluateFunction(x) << endl;

  return EXIT_SUCCESS;
}
